src/DB/Query/ShippingRateQuery.php

```diff
@@ -60,7 +60,7 @@ protected function query_results() {
 
 		$this->results = array_map(
 			function ( $row ) {
-				$row['options'] = maybe_unserialize( $row['options'] );
+				$row['options'] = ! empty( $row['options'] ) ? json_decode( $row['options'] ) : $row['options'];
```

Comment:
I think we need to be careful with this conversion if we aren't going to pass `$associative = true`. It's not a problem since we pass it directly to the rest response which will convert either an object or an associative array to the same JSON.

In the controller tests we are mocking the options data as an associative array, so that's not really clear. I'm fine with leaving it either way, but we'd need to clarify that somewhere. It could also become a problem if we start validating the schema multiple levels deep.

Reply:
Yeah it'd be better just to decode them as associative arrays: https://github.com/woocommerce/google-listings-and-ads/pull/1305/commits/ec5745750e6118e29589bd349d2dbdc9df00a110